### ISSUE TITLE
Failed to build ImageMagick because another Homebrew process is already using xz

### ISSUE THREAD

**Issue by Gnimuc** [JuliaIO/ImageMagick.jl#47]
I've found some clues in these issues: [legacy-homebrew#47593](https://github.com/Homebrew/legacy-homebrew/issues/47593), [legacy-homebrew#17730](https://github.com/Homebrew/legacy-homebrew/issues/17730) and [homebrew-julia#21](https://github.com/staticfloat/homebrew-julia/issues/21), but none of them gives an effective solution. 

This is what I've tried:
- brew update
- brew doctor  => Your system is ready to brew.
- brew cleanup
- reboot OS
- cleanup all `.brewing` files  in `$(brew --cache)/Formula/`
- brew rm -f xz
- rm `$(brew missing xz)`

Any help will be appreciated!

``` julia
INFO: Building ImageMagick
==> Installing imagemagick from staticfloat/juliatranslated
Error: Operation already in progress for xz
Another active Homebrew process is already using xz.
Please wait for it to finish or terminate it to continue.
=============================[ ERROR: ImageMagick ]=============================

LoadError: failed process: Process(`/Users/gnimuc/.julia/v0.4/Homebrew/deps/usr/bin/brew install staticfloat/juliatranslated/imagemagick`, ProcessExited(1)) [1]
while loading /Users/gnimuc/.julia/v0.4/ImageMagick/deps/build.jl, in expression starting on line 98

================================================================================

=============================================================[ BUILD ERRORS ]==============================================================

WARNING: ImageMagick had build errors.

 - packages with build errors remain installed in /Users/gnimuc/.julia/v0.4
 - build the package(s) and all dependencies with `Pkg.build("ImageMagick")`
 - build a single package by running its `deps/build.jl` script

===========================================================================================================================================

julia> versioninfo()
Julia Version 0.4.5
Commit 2ac304d (2016-03-18 00:58 UTC)
Platform Info:
  System: Darwin (x86_64-apple-darwin13.4.0)
  CPU: Intel(R) Core(TM) i5-3210M CPU @ 2.50GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Sandybridge)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.3
```

**Comment by kmsquire** [comment]
Cc: @staticfloat

**Comment by Gnimuc** [comment]
I can successfully build the package on a brand new system, here are some build information:

``` julia
INFO: Building ImageMagick
WARNING: The following packages do not have relocatable bottles, installation may fail!
Please report these packages to https://github.com/JuliaLang/Homebrew.jl:
  libpng
  freetype
  jpeg
  imagemagick
Warning: You are using OS X 10.12.
We do not provide support for this pre-release version.
You may encounter build failures or other breakages.
Please create pull-requests instead of filing issues.
==> Installing dependencies for libpng: xz
==> Installing libpng dependency: xz
==> Downloading https://fossies.org/linux/misc/xz-5.2.2.tar.gz
######################################################################## 100.0%
==> ./configure --disable-silent-rules --prefix=/Users/gnimuc/.julia/v0.4/Homebrew/deps/usr/Cellar/x
==> make install
🍺  /Users/gnimuc/.julia/v0.4/Homebrew/deps/usr/Cellar/xz/5.2.2: 91 files, 1.4M, built in 1 minute 37 seconds
==> Installing libpng
==> Downloading ftp://ftp.simplesystems.org/pub/libpng/png/src/libpng16/libpng-1.6.23.tar.xz
.
.
.
==> ./configure --prefix=/Users/gnimuc/.julia/v0.4/Homebrew/deps/usr/Cellar/libtiff/4.0.6_2 --without-x --with-jpeg-include-dir=/Users/gnimuc
==> make install
🍺  /Users/gnimuc/.julia/v0.4/Homebrew/deps/usr/Cellar/libtiff/4.0.6_2: 261 files, 3.4M, built in 1 minute 30 seconds
==> Installing imagemagick
==> Downloading https://dl.bintray.com/homebrew/mirror/imagemagick-6.9.5-7.tar.xz
######################################################################## 100.0%
==> ./configure --disable-osx-universal-binary --prefix=/Users/gnimuc/.julia/v0.4/Homebrew/deps/usr/Cellar/imagemagick/6.9.5-7 --disable-sile
==> make install
🍺  /Users/gnimuc/.julia/v0.4/Homebrew/deps/usr/Cellar/imagemagick/6.9.5-7: 1,464 files, 22.4M, built in 8 minutes 32 seconds
Warning: Already linked: /Users/gnimuc/.julia/v0.4/Homebrew/deps/usr/Cellar/imagemagick/6.9.5-7
INFO: Package database updated
```

**Comment by staticfloat** [comment]
@Gnimuc Your latest post shows that things are working for you, but they're not working like we'd want them to, because you're having to download everything from source and compile your own binaries.  This is a fallback that we never really want you to have to deal with, unfortunately it's necessary because of the big warning at the top of your log:

```
Warning: You are using OS X 10.12.
We do not provide support for this pre-release version.
You may encounter build failures or other breakages.
```

This should be addressed shortly after macOS 10.12 is released, so until then, I'm sorry but you will likely be plagued with dependency issues.
